pandas/tests/frame/methods/test_astype.py

```diff
@@ -428,11 +428,23 @@ def test_astype_to_incorrect_datetimelike(self, unit):
         other = f"m8[{unit}]"
 
         df = DataFrame(np.array([[1, 2, 3]], dtype=dtype))
-        msg = fr"Cannot cast DatetimeArray to dtype timedelta64\[{unit}\]"
+        msg = "|".join(
+            [
+                fr"Cannot cast DatetimeArray to dtype timedelta64\[{unit}\]",
+                "cannot astype a datetimelike from "
+                fr"\[datetime64\[ns\]\] to \[timedelta64\[{unit}\]\]",
+            ]
+        )
         with pytest.raises(TypeError, match=msg):
             df.astype(other)
 
-        msg = fr"Cannot cast TimedeltaArray to dtype datetime64\[{unit}\]"
+        msg = "|".join(
+            [
+                fr"Cannot cast TimedeltaArray to dtype datetime64\[{unit}\]",
+                "cannot astype a timedelta from "
```

Comment:
not sure if its worth the trouble, but on several recent PRs ive been wondering if we should comment these as being BM/AM-specific

Reply:
It could be useful to clean it up in the future, yes, will add a comment.